File: src/main/java/xyz/emirdev/echogen/commands/EnderChestCommand.java
package xyz.emirdev.echogen.commands;

import org.bukkit.entity.Player;

import com.mojang.brigadier.context.CommandContext;
import com.mojang.brigadier.exceptions.CommandSyntaxException;
import com.mojang.brigadier.tree.LiteralCommandNode;

import io.papermc.paper.command.brigadier.CommandSourceStack;
import io.papermc.paper.command.brigadier.Commands;
import io.papermc.paper.command.brigadier.argument.ArgumentTypes;
import xyz.emirdev.echogen.PluginCommand;

import java.util.Set;

public class EnderChestCommand extends PluginCommand {

    @Override
    public LiteralCommandNode<CommandSourceStack> getCommand() {
        return Commands.literal("enderchest")
                .requires(hasPermission("echogen.enderchest"))
                .executes(this::enderchest)
                .then(Commands.argument("player", ArgumentTypes.player())
                        .requires(hasPermission("echogen.enderchest.others"))
                        .executes(this::enderchestOther))
                .build();
    }

    @Override
    public Set<String> getAliases() {
        return Set.of("ec");
    }

    private int enderchest(CommandContext<CommandSourceStack> ctx) {
        Player player = getContextPlayer(ctx);
        if (player == null)
            return 1;

        player.openInventory(player.getEnderChest());

        return 1;
    }

    private int enderchestOther(CommandContext<CommandSourceStack> ctx) throws CommandSyntaxException {
        Player player = getContextPlayer(ctx);
        if (player == null)
            return 1;

        Player target = getPlayer(ctx, "player");

        player.openInventory(target.getEnderChest());

        return 1;
    }
}

File: src/main/kotlin/cat/emir/echogen/commands/EnderChestCommand.kt
package cat.emir.echogen.commands

import cat.emir.echogen.commandlib.PluginCommand
import cat.emir.echogen.commandlib.getPlayer
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.mojang.brigadier.context.CommandContext
import io.papermc.paper.command.brigadier.CommandSourceStack

import io.papermc.paper.command.brigadier.argument.ArgumentTypes

class EnderChestCommand : PluginCommand() {
    override val aliases = setOf("ec")

    override fun getCommand(): LiteralArgumentBuilder<CommandSourceStack> {
        return command("enderchest") {
            requires { it.sender.hasPermission("echogen.enderchest") }
            executes(::enderchest)
            argument("player", ArgumentTypes.player()) {
                requires { it.sender.hasPermission("echogen.enderchest.others") }
                executes(::enderchestOther)
            }
        }
    }

    fun enderchest(ctx: CommandContext<CommandSourceStack>): Int {
        val player = ctx.getPlayer() ?: return 1

        player.openInventory(player.enderChest)

        return 1
    }

    fun enderchestOther(ctx: CommandContext<CommandSourceStack>): Int {
        val player = ctx.getPlayer() ?: return 1
        val target = ctx.getPlayer("player") ?: return 1

        player.openInventory(target.enderChest)

        return 1
    }
}
